internal/provider/helper_test.go

```diff
@@ -322,6 +322,28 @@ func testAccAddProjectMembers(t *testing.T, pid interface{}, users []*gitlab.Use
 	}
 }
 
+func testAccCreateProjectHooks(t *testing.T, pid interface{}, n int) []*gitlab.ProjectHook {
+	t.Helper()
+
+	var hooks []*gitlab.ProjectHook
+	for i := 0; i < n; i++ {
+		hook, _, err := testGitlabClient.Projects.AddProjectHook(pid, &gitlab.AddProjectHookOptions{
+			URL: gitlab.String(fmt.Sprintf("https://%s.com", acctest.RandomWithPrefix("acctest"))),
+		})
+		if err != nil {
+			t.Fatalf("could not create project hook: %v", err)
+		}
+		t.Cleanup(func() {
+			_, err := testGitlabClient.Projects.DeleteProjectHook(pid, hook.ID)
+			if err != nil {
+				t.Fatalf("could not cleanup project hook: %v", err)
+			}
+		})
```

Comment:
@timofurrer  - Do we have a weird potential race condition here with the executed order of the cleanup scripts? In most of our project-bound resources in the `helper_test` class we don't clean them up because we allow the deletion of the project to clean them up (see protected branches, project members, etc); This makes sense because it could cause an ordering issue if they're trying to clean them up after the project was already deleted. the `t.Fatalf` block would be hit here if the project was cleaned up before the hooks were, right?

Seems like in this case we'd be OK to leave off the cleanup method specifically because they'd be cleaned up when the project was.

Reply:
Yes, indeed - I've removed it. We actually we have the same potential issue in other helpers ...

Reply:
Yeah, I saw a couple there with the same issue. I may open a PR to see about closing that up, even though it seems to be pretty rare.